### PR TITLE
feat: adds Container instrumentation tests

### PR DESCRIPTION
**Pull request by carlossteinzup**
### Related Issues

https://github.com/ZupIT/beagle/issues/500
https://github.com/ZupIT/beagle/issues/1019

### Description and Example

Thi PR adds the instrumentation tests to the container component.

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/ContainersController.kt [hunk 1]
```diff
@@ -16,14 +16,12 @@
 
 package br.com.zup.beagle.automatedtests.controllers
 
+import br.com.zup.beagle.automatedtests.builders.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/ContainersController.kt:19]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d4cd0b8e7cd74fa103693bb11a73c089e8f1170/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/ContainersController.kt#L19"></span>br.com.zup.beagle.automatedtests.builders.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: android/automated-tests/build.gradle [hunk 2]
```diff
@@ -89,8 +89,13 @@ dependencies {
     androidTestImplementation Dependencies.TestLibraries.testRules
     androidTestImplementation Dependencies.TestLibraries.testRunner
     androidTestImplementation Dependencies.TestLibraries.testExt
+    implementation "androidx.core:core-ktx:+"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 
 }
+repositories {
+    mavenCentral()
+}
```
Comment (uziasferreirazup) [inline android/automated-tests/build.gradle:98]: delete these lines, this has already been added just see the file in full

File: iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import XCTest
+
+class ContainerSteps: CucumberStepsDefinition {
+    var application: XCUIApplication!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift:21]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d4cd0b8e7cd74fa103693bb11a73c089e8f1170/iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift#L21"></span>Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>ContainerSteps.swift:21</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        // MARK: - Given
+        Given("^the Beagle application did launch with the container screen url$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.CONTAINER_SCREEN_TITLE.element.exists)
+        }
+        
+        // MARK: - Then
+        
+        // Scenario 1
+        Then("^the screen should contain three texts: \"([^\\\"]*)\", \"([^\\\"]*)\" and \"([^\\\"]*)\"$") { args, _ -> Void in
+            let texts = [args![0], args![1], args![2]]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift:41]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d4cd0b8e7cd74fa103693bb11a73c089e8f1170/iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift#L41"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>ContainerSteps.swift:41</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            for text in texts {
+                XCTAssertTrue(self.application.staticTexts[text].exists)
+            }
+        }
+        
+        // Scenario 2
+        Then("^the text \"([^\\\"]*)\" and the text \"([^\\\"]*)\" set via context should be displayed$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift:49]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d4cd0b8e7cd74fa103693bb11a73c089e8f1170/iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift#L49"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>ContainerSteps.swift:49</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+        }
+        
+        // Scenario 3
+        Then("^the text \"([^\\\"]*)\" and the text \"([^\\\"]*)\" set via context should be displayed$") { args, _ -> Void in
+            let texts = [args![0], args![1]]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift:55]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d4cd0b8e7cd74fa103693bb11a73c089e8f1170/iOS/Tests/AutomatedTestsUITests/Steps/ContainerSteps.swift#L55"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>ContainerSteps.swift:55</code></td>
    </tr>
  </tbody>
</table>

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/ContainerSteps.kt [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    @Then("^the views that contains the text (.*) set via context must be displayed$")
+    fun checkTextExistsInAViewSetViaContext(string1:String) {
+        ScreenRobot()
+            .checkViewContainsText(string1, true)
+    }
+
+
+
+
+//
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/ContainerSteps.kt:72]: is there any reason why the lines below are commented?
I recommend don't comment code, just don't commit theses lines and create an issue reference about this.

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt [hunk 8]
```diff
@@ -30,10 +30,10 @@ class Instrumentation : MonitoringInstrumentation() {
...
     override fun onCreate(arguments: Bundle) {
         super.onCreate(arguments)
-        val tags: String = BuildConfig.TEST_TAGS
-        if (tags.isNotEmpty()) {
-            arguments.putString("tags", tags.replace(",".toRegex(), "--").replace("\\s".toRegex(), ""))
-        }
+//        val tags: String = BuildConfig.TEST_TAGS
+//        if (tags.isNotEmpty()) {
+//            arguments.putString("tags", tags.replace(",".toRegex(), "--").replace("\\s".toRegex(), ""))
+//        }
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt:36]: there is some reason to comment these lines?

File: android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/MainActivity.kt [hunk 9]
```diff
@@ -40,5 +40,5 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    fun bffUrl() = intent.extras?.getString(BFF_URL_KEY) ?: "http://10.0.2.2:8080/navigate-actions"
+    fun bffUrl() = intent.extras?.getString(BFF_URL_KEY) ?: "http://10.0.2.2:8080/container-test"
```
Comment (uziasferreirazup) [inline android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/MainActivity.kt:43]: always add new test this `bffurl` change I recommend keep the base URL and never commit the change.

File: android/beagle/build.gradle [hunk 10]
```diff
@@ -93,6 +93,11 @@ dependencies {
...
     testImplementation Dependencies.TestLibraries.archCoreTesting
     testImplementation Dependencies.GeneralLibraries.jsonObject
+    implementation "androidx.core:core-ktx:+"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 }
 
 apply from: rootProject.file('../maven-publish.gradle')
+repositories {
+    mavenCentral()
+}
```
Comment (uziasferreirazup) [inline android/beagle/build.gradle:103]: delete these lines, this has already been added just see the file in full

File: android/build.gradle [hunk 11]
```diff
@@ -15,6 +15,7 @@
  */
 
 buildscript {
+    ext.kotlin_version = '1.4.10'
```
Comment (uziasferreirazup) [inline android/build.gradle:18]: delete these lines, this has already been added just see the file in full

File: android/build.gradle [hunk 12]
```diff
@@ -28,6 +29,7 @@ buildscript {
         classpath "org.jetbrains.dokka:dokka-gradle-plugin:1.4.10"
         classpath 'com.hiya:jacoco-android:0.2'
         classpath 'com.vanniktech:gradle-maven-publish-plugin:0.13.0'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
```
Comment (uziasferreirazup) [inline android/build.gradle:32]: delete these lines, this has already been added just see the file in full

File: android/preview/build.gradle [hunk 13]
```diff
@@ -58,6 +58,11 @@ dependencies {
...
     testImplementation Dependencies.TestLibraries.kotlinTest
     testImplementation Dependencies.TestLibraries.mockk
+    implementation "androidx.core:core-ktx:+"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 }
 
 apply from: rootProject.file('../maven-publish.gradle')
+repositories {
+    mavenCentral()
+}
```
Comment (uziasferreirazup) [inline android/preview/build.gradle:68]: delete these lines, this has already been added just see the file in full

File: android/sample/build.gradle [hunk 14]
```diff
@@ -65,4 +65,9 @@ dependencies {
 
     implementation project(Dependencies.Modules.core)
     kapt project(Dependencies.Modules.processor)
+    implementation "androidx.core:core-ktx:+"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+}
+repositories {
+    mavenCentral()
 }
```
Comment (uziasferreirazup) [inline android/sample/build.gradle:73]: delete these lines, this has already been added just see the file in full

**Review by uziasferreirazup — CHANGES_REQUESTED** [review]

File: android/automated-tests/build.gradle [hunk 15]
```diff
@@ -89,8 +89,13 @@ dependencies {
     androidTestImplementation Dependencies.TestLibraries.testRules
     androidTestImplementation Dependencies.TestLibraries.testRunner
     androidTestImplementation Dependencies.TestLibraries.testExt
+    implementation "androidx.core:core-ktx:+"
```
Comment (botbeaglezup) [inline android/automated-tests/build.gradle:92]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/07ee3b985dabea1a15252e8d15b996f9fcc6e94b/android/automated-tests/build.gradle#L92"></span>Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (androidx.core:core-ktx:+)</td>
    </tr>
  </tbody>
</table>

File: android/beagle/build.gradle [hunk 16]
```diff
@@ -93,6 +93,11 @@ dependencies {
     testImplementation Dependencies.TestLibraries.mockk
     testImplementation Dependencies.TestLibraries.archCoreTesting
     testImplementation Dependencies.GeneralLibraries.jsonObject
+    implementation "androidx.core:core-ktx:+"
```
Comment (botbeaglezup) [inline android/beagle/build.gradle:96]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/07ee3b985dabea1a15252e8d15b996f9fcc6e94b/android/beagle/build.gradle#L96"></span>Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (androidx.core:core-ktx:+)</td>
    </tr>
  </tbody>
</table>

File: android/preview/build.gradle [hunk 17]
```diff
@@ -58,6 +58,11 @@ dependencies {
     testImplementation Dependencies.TestLibraries.junit
     testImplementation Dependencies.TestLibraries.kotlinTest
     testImplementation Dependencies.TestLibraries.mockk
+    implementation "androidx.core:core-ktx:+"
```
Comment (botbeaglezup) [inline android/preview/build.gradle:61]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/07ee3b985dabea1a15252e8d15b996f9fcc6e94b/android/preview/build.gradle#L61"></span>Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (androidx.core:core-ktx:+)</td>
    </tr>
  </tbody>
</table>

File: android/sample/build.gradle [hunk 18]
```diff
@@ -65,4 +65,9 @@ dependencies {
 
     implementation project(Dependencies.Modules.core)
     kapt project(Dependencies.Modules.processor)
+    implementation "androidx.core:core-ktx:+"
```
Comment (botbeaglezup) [inline android/sample/build.gradle:68]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/07ee3b985dabea1a15252e8d15b996f9fcc6e94b/android/sample/build.gradle#L68"></span>Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (androidx.core:core-ktx:+)</td>
    </tr>
  </tbody>
</table>

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt (hunk 8 above)
Comment (carlossteinzup) [reply android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt]: Yes. The the app was requesting to reeimport the BuildConfig everytime it runs. Then I ask Pó about and he said we could comment that to check later how to fix it. I have removed the lines and will create an issue for it.

File: android/automated-tests/build.gradle (hunk 2 above)
Comment (carlossteinzup) [reply android/automated-tests/build.gradle:98]: done

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/ContainerSteps.kt (hunk 7 above)
Comment (carlossteinzup) [reply android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/ContainerSteps.kt]: removed
done

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt (hunk 8 above)
Comment (carlossteinzup) [reply android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Instrumentation.kt]: done

File: android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/MainActivity.kt (hunk 9 above)
Comment (carlossteinzup) [reply android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/MainActivity.kt]: ok. we will set the container test screen as default

File: android/beagle/build.gradle (hunk 10 above)
Comment (carlossteinzup) [reply android/beagle/build.gradle:103]: done

File: android/build.gradle (hunk 12 above)
Comment (carlossteinzup) [reply android/build.gradle:32]: done

File: android/preview/build.gradle (hunk 13 above)
Comment (carlossteinzup) [reply android/preview/build.gradle:68]: done

File: android/sample/build.gradle (hunk 14 above)
Comment (carlossteinzup) [reply android/sample/build.gradle:73]: done

**Review by uziasferreirazup — APPROVED** [review]

**Review by tiagomnunes89 — APPROVED** [review]

File: android/preview/build.gradle [hunk 19]
```diff
@@ -61,3 +61,4 @@ dependencies {
 }
 
 apply from: rootProject.file('../maven-publish.gradle')
+
```
Comment (viniciusguardieirozup) [inline android/preview/build.gradle:64]: please, remove this blank space

**Review by viniciusguardieirozup — APPROVED** [review]

**Review by caioortu — APPROVED** [review]

**Pull request merged**